core/src/test/java/org/apache/iceberg/deletes/TestRoaringPositionBitmap.java

```diff
@@ -0,0 +1,516 @@
+      long minInclusive, long maxExclusive, int size) {
+    Random random = new Random(seed);
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+    Set<Long> positions = Sets.newHashSet();
+
+    while (positions.size() < size) {
+      long position = nextLong(random, minInclusive, maxExclusive);
+      positions.add(position);
+      bitmap.add(position);
+    }
+
+    return Pair.of(bitmap, positions);
+  }
+
+  private Pair<RoaringPositionBitmap, Set<Long>> generateDenseBitmap(int requiredBitmapCount) {
+    Random random = new Random(seed);
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+    Set<Long> positions = Sets.newHashSet();
+    long currentPosition = 0;
+
+    while (bitmap.allocatedBitmapCount() <= requiredBitmapCount) {
+      long maxRunPosition = currentPosition + nextLong(random, 1000, 2 * CONTAINER_SIZE);
+      for (long position = currentPosition; position <= maxRunPosition; position++) {
+        bitmap.add(position);
+        positions.add(position);
+      }
+      long shift = nextLong(random, (long) (0.1 * BITMAP_SIZE), (long) (0.25 * BITMAP_SIZE));
+      currentPosition = maxRunPosition + shift;
+    }
+
+    return Pair.of(bitmap, positions);
+  }
+
+  private void assertRandomPositions(RoaringPositionBitmap bitmap, Set<Long> positions) {
+    Random random = new Random(validationSeed);
+    for (int ordinal = 0; ordinal < VALIDATION_LOOKUP_COUNT; ordinal++) {
+      long position = nextLong(random, 0, RoaringPositionBitmap.MAX_POSITION);
+      assertThat(bitmap.contains(position)).isEqualTo(positions.contains(position));
+    }
+  }
+
+  private static long nextLong(Random random, long minInclusive, long maxExclusive) {
+    return minInclusive + (long) (random.nextDouble() * (maxExclusive - minInclusive));
+  }
+
+  private static long position(int bitmapIndex, int containerIndex, long value) {
+    return bitmapIndex * BITMAP_OFFSET + containerIndex * CONTAINER_OFFSET + value;
+  }
+
+  private static RoaringPositionBitmap roundTripSerialize(RoaringPositionBitmap bitmap) {
+    ByteBuffer buffer = ByteBuffer.allocate((int) bitmap.serializedSizeInBytes());
+    buffer.order(ByteOrder.LITTLE_ENDIAN);
+    bitmap.serialize(buffer);
+    buffer.flip();
+    return RoaringPositionBitmap.deserialize(buffer);
+  }
+
+  private static RoaringPositionBitmap readBitmap(String resourceName) throws IOException {
+    byte[] bytes = readTestResource(resourceName);
+    ByteBuffer buffer = ByteBuffer.wrap(bytes);
+    buffer.order(ByteOrder.LITTLE_ENDIAN);
+    return RoaringPositionBitmap.deserialize(buffer);
+  }
+
+  private static byte[] readTestResource(String resourceName) throws IOException {
+    URL resource = Resources.getResource(TestRoaringPositionBitmap.class, resourceName);
+    return Resources.toByteArray(resource);
+  }
+
+  private static void assertEqual(RoaringPositionBitmap bitmap, Set<Long> positions) {
```

Comment:
this is really 3 asserts
assertEqual()
assertEqualAfterSerDE()
assertEqualAfterCompress()

, but it's probably fine to combine them since the debugger will point to which block we are in

Reply:
I added a common method for the checks.